tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -42,6 +53,51 @@ def test_run_updates_task_metadata(k8s_executor):
     )
 
 
+def test_run(k8s_executor):
+    task_config = KubernetesTaskConfig(
+        name="fake_task_name",
+        uuid="fake_id",
+        image="fake_docker_image",
+        command="fake_command"
+    )
+    fake_container = V1Container(
+        image=task_config.image,
+        name=task_config.name,
+        command=[task_config.command]
+    )
+    fake_pod = V1Pod(
+        metadata=V1ObjectMeta(
+            name=task_config.pod_name,
+            namespace="task_processing_tests"
+        ),
+        spec=V1PodSpec(
+            restart_policy=task_config.restart_policy,
+            containers=[fake_container]
+        ),
+    )
+
+    assert k8s_executor.run(task_config) == task_config.pod_name
+    assert k8s_executor.kube_client.core.create_namespaced_pod.call_args_list == [
+        mock.call(body=fake_pod, namespace='task_processing_tests')
+    ]
+
+
+@patch("task_processing.plugins.kubernetes.kubernetes_pod_executor.logger", autospec=True)
+def test_run_failed_exception(mock_logger, k8s_executor):
+    task_config = KubernetesTaskConfig(
+        name="fake_task_name",
+        uuid="fake_id",
+        image="fake_docker_image",
+        command="fake_command"
+    )
+    k8s_executor.kube_client.core.create_namespaced_pod.side_effect = ApiException(
+        status=403, reason="Fake unauthorized message")
+    assert k8s_executor.run(task_config) is None
+    assert mock_logger.exception.call_args(mock.call(
+        f"Failed to create pod {task_config.pod_name}:",
+        f"{k8s_executor.kube_client.core.create_namespaced_pod.side_effect}"))
```

Comment:
its probably fine to not test that we logged something - this test will break if we ever change the wording even tho no functionality has changed
```suggestion
```